src/answer/precious.rs

```diff
@@ -256,11 +261,18 @@ fn parse_answer_detailed(
         } else {
             let mut formatted_answer: String = String::new();
             for sub_node in instruction.children() {
+                println!("{:?}", sub_node);
+                let syntax: Option<String> = if let Some(s) = sub_node.attr("class") {
+                    class_string_syntax(s.to_string())
+                } else { None };
+
                 match sub_node.name() {
-                    Some("pre") => formatted_answer
-                        .push_str(&(colorized_code(sub_node.text(), &question_tags) + "\n")),
+                    Some("pre") => {
+                        formatted_answer
+                            .push_str(&(colorized_code(sub_node.text(), &question_tags, syntax) + "\n"))
```

Comment:
I think it's better to introduce a new function like this:
```rust
/// docstring
fn extract_code_colorized(code_node: Node, possible_tags: &[String]) -> String {
    println!("debug {:?}", code_node.attr("class"));
    let syntax = code_node.attr("class").and_then(class_string_syntax);
    colorized_code(code_node.text(), possible_tags, syntax)
}
```

So we can avoid  duplicate code in  in both `parse_answer_instruction` function and `parse_answer_detailed`